tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/aux/caches/IntValueCache.scala

```diff
@@ -0,0 +1,38 @@
+package at.forsyte.apalache.tla.bmcmt.stratifiedRules.aux.caches
+
+import at.forsyte.apalache.tla.bmcmt.smt.SolverContext
+import at.forsyte.apalache.tla.bmcmt.types.CellT
+import at.forsyte.apalache.tla.bmcmt.{ArenaCell, PureArena}
+import at.forsyte.apalache.tla.lir.IntT1
+import at.forsyte.apalache.tla.types.tla
+
+/**
+ * A cache for integer constants that maps an integer to an integer constant in SMT.
+ *
+ * @author
+ *   Jure Kukovec
+ */
+class IntValueCache extends Cache[PureArena, BigInt, ArenaCell] {
+
+  def create(arena: PureArena, intValue: BigInt): (PureArena, ArenaCell) = {
+    // introduce a new constant
+    val newArena = arena.appendCell(CellT.fromType1(IntT1))
+    val intCell = newArena.topCell
+    (newArena, intCell)
+  }
```

Comment:
According to docs on the trait, this should be caching the source, right? But I don't see any cache entry happening here. I'm I missing something?

Reply:
`getOrCreate` should be the only entry point, and `addToCache` is called there, one should never be calling `create` directly. Looking at it, I'm not 100% sure what the public-ness of a method that is declared as `protected` in `Cache` but def here is, but I would assume `protected` too.

Reply:
Apologies if I am missing something, but `create` is defined as an abstract method on the the parent class, and is documented to say `Create a target value based on the source value and cache it.` Here you are implementing the method, but I don't see it caching. May that comment is stale?

Reply:
You're referencing a comment from an old version, it has since been updated to "Compute a target value and context update, based on the source value, but do not save the result."